### PR TITLE
swamp: add options to swamp config

### LINKED ISSUE #727: swamp: add the ability to pass a custom config to swamp creation
We should have the ability to pass a custom options to swamp's config

### PR DESCRIPTION
**Pull request by vgonkivs**
Resolves #727

### CODE REVIEW

**Review by Wondertan — APPROVED** [review]
LGTM. I have a few options on my mind that I will add later on. Thank you!

File: node/tests/swamp/config.go [hunk 1]
```diff
@@ -26,3 +26,13 @@ func DefaultComponents() *Components {
 		CoreCfg: tnCfg,
 	}
 }
+
+// Option for Swamp's Config.
+type Option func(*Components)
+
+// WithBlockInterval sets custom interval for blocks creation
```
Comment (Wondertan) [inline node/tests/swamp/config.go:33]:
```suggestion
// WithBlockInterval sets a custom interval for block creation.
```

File: node/tests/swamp/config.go [hunk 2]
```diff
@@ -26,3 +26,13 @@ func DefaultComponents() *Components {
 		CoreCfg: tnCfg,
 	}
 }
+
+// Option for Swamp's Config.
```
Comment (renaynay) [inline node/tests/swamp/config.go:30]:
```suggestion
// Option for modifying Swamp's Config.
```

**Review by renaynay — APPROVED** [review]

**Review by Bidon15 — APPROVED** [review]
Awesome 👏 
🎬 🚢

**Pull request merged**